Karl/test_service/annotator.h

```diff
@@ -55,43 +56,28 @@ class ServiceAnnotator final {
     // In example "production" mode just start regular keepalives.
     claire_.Register();
 #endif
-  }
-
-  ~ServiceAnnotator() {
-    destructing_ = true;
-    thread_.join();
+    claire_.AddDependency(service_generator);
```

Comment:
Aren't these initialized by an initializer list in Claire's construction in the initialization list?